### PR TITLE
SITES-79 [Sites30] Proxy to open the site

### PR DESCRIPTION
**Pull request by richardhand**
- open the site directly
- adds gitignore line for intellij projects

### CODE REVIEW

**Review by jckautzmann — APPROVED** [review]

File: lib/browserSync.js [hunk 1]
```diff
@@ -9,10 +9,10 @@ it.
 
 const browserSync = require('browser-sync').create();
 
-function runBrowserSync ({ port, portProxy }) {
+function runBrowserSync ({ port, portProxy, siteName }) {
```
Comment (bartoszglow) [inline lib/browserSync.js:12]: I think `siteName` could have default value set. That way you don't have to pass it in the bin script as an argument.

File: bin/index.js [hunk 2]
```diff
@@ -13,6 +13,8 @@ const shell = require('shelljs');
 const proxy = require('../lib/proxy');
 const browserSync = require('../lib/browserSync');
 const findPort = require('../lib/findPort');
+const dotenv = require('dotenv');
+const env = dotenv.config().parsed;
```
Comment (bartoszglow) [inline bin/index.js:17]: WDYT about extending proxyConfig by siteName?

File: lib/browserSync.js [hunk 3]
```diff
@@ -9,10 +9,10 @@ it.
...
   browserSync.init({
     port,
-    proxy: `localhost:${portProxy}`,
+    proxy: `localhost:${portProxy}/content/${siteName}.html`,
```
Comment (bartoszglow) [inline lib/browserSync.js:15]: Do we want to always open this link? Is there any chance to standardize behavior with the "open" button after page creation?

Comment (richardhand) [reply lib/browserSync.js:15]: I think this is already good start, but yes it's likely you would want be able to configure:
- whether or not to open in edit or viewAsPublished
- the start page to open

so basically the same as we implemented in the open action after site creation. Unfortunately we're detached here from the template configuration and cannot reuse it as far as I can tell.

File: bin/index.js (hunk 2 above)
Comment (richardhand) [reply bin/index.js:17]: I left it out as to me it isn't really a configuration for the proxy, rather just an environment config

File: lib/browserSync.js (hunk 3 above)
Comment (bartoszglow) [reply lib/browserSync.js:15]: we could optionally export additional vars for "download theme sources" as we do already interact with those then we could mimic the same behavior in proxy.

File: lib/browserSync.js (hunk 1 above)
Comment (bartoszglow) [reply lib/browserSync.js:12]: the default value should point to `env.AEM_SITE`

File: lib/browserSync.js (hunk 3 above)
Comment (richardhand) [reply lib/browserSync.js]: thinking more about it, reusing the site template configuration seems like an undesired/hidden behaviour. The `openItem` configuration in the site template can be custom and for example, used to open a second configuration url post site creation. In that case you would not want to reuse such a url in the proxy.

I would rather have a simple way of configuring similar properties in the theme builder (env variable, flag etc.). For now, let's keep this simple and discuss this idea in the review next week.

File: lib/browserSync.js [hunk 4]
```diff
@@ -9,10 +9,10 @@ it.
...
     files: 'dist'
```
[the remark this answers is not in the document]
Comment (richardhand) [reply lib/browserSync.js]: added in 7352b4a50ec7e3

File: lib/browserSync.js (hunk 1 above)
Comment (richardhand) [reply lib/browserSync.js]: removed in 7352b4a50ec7e3

**Review by bartoszglow — APPROVED** [review]

**Pull request merged**